CMakeLists.txt

```diff
@@ -5,6 +5,10 @@ if(NOT DEFINED PROJECT_NAME)
     set(CPP_INDIRECT_IS_SUBPROJECT ON)
 endif()
 
+# TODO(Avoid gloablly setting C++ standard) Force C++17 for Catchv2
```

Comment:
This should be fine, as it will only set the C++ standard for our test executable and not have any effect on user building against the generated config module.

This may not be the case for users, including via FetchContent. In that case, we should probably just wrap this with an "if indirect_value is the top level project".

Reply:
So I've raised https://github.com/jbcoe/indirect_value/pull/121/files to address the fact that `CPP_INDIRECT_IS_SUBPROJECT` should be called `CPP_INDIRECT_IS_NOT_SUBPROJECT`.  Once corrected, this check should go within this check, preferably right after the point we fetch catch so it's logically scoped to where we want it to apply.  Line 61 after we fetch Catch2 makes the most sense I think.

Reply:
Are you happy for me to merge as-is or do you want your PR in first?

Reply:
I've moved this into the test-only code to avoid affecting library code (when included via add_subdirectory`).  

Its ready to commit now.